src/Plot.tsx

```diff
@@ -247,6 +249,7 @@ export class Plot extends Layout {
     context.fillStyle = resolveCanvasStyle(this.xAxisTextColor(), context);
     context.font = `${this.labelSize().y}px ${this.fontFamily()}`;
     context.textAlign = 'center';
+    context.textBaseline = 'alphabetic';
```

Comment:
TIL about `alphabetic`. I think the same thing applies here for pixel-pushing, where in some edge configurations the descenders might go slightly out of bounds, but I don't know that that's worth solving for.